### PR TITLE
Bug 1314864 - Fix crashes in SyncView and document sync_summary

### PR DESCRIPTION
**Pull request by thomcc**
It's a little disappointing to go back to the deprecated method in the first commit but AFAICT it's necessary.

Let me know if anything else is needed here.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/telemetry-batch-view/149)

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by thomcc** [comment]
There's no specific bug number on bugzilla for these, err, bugfixes, so I'm using the original bug which is still open.

CC @mhammond

**Comment by vitillo** [comment]
@harterrt r?

**Comment by harterrt** [comment]
> It's a little disappointing to go back to the deprecated method in the first commit but AFAICT it's necessary.

Can you expand on why it's necessary? Is it erroring out?

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/main/scala/com/mozilla/telemetry/views/SyncView.scala, line 474 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149#-KXuvmfS5x7fdg2b56tN:-KXuvmfS5x7fdg2b56tO:bxdszor) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/4ba5c1ab150a3c223e853684b376b730874246e3/src/main/scala/com/mozilla/telemetry/views/SyncView.scala#L474)):*
> ```Scala
>       stringFromSyncOrPayload("uid") match {
>         case null => return None // required.
>         case x => x
> ```

Please change this comment to match the others, i.e.: s/required/a required field/. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149#-:-KXuyf3XVgRxYuNM_hh3:b77bw8v)*

**Comment by thomcc** [comment]
It causes the flatMap closure below to be unserializable (for reasons I don't really understand -- ISTM like LongAccumulator *should* be serializable...)

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149#-:-KXuzAhNfAObXNktFiBO:b1wo5sc)*

**Comment by harterrt** [comment]
One nit below. Otherwise, :lgtm: 

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149#-:-KXv7swgNSXQX0j4C9bN:bu0rvu1)*

File: src/main/scala/com/mozilla/telemetry/views/SyncView.scala [hunk 1]
```diff
@@ -63,8 +63,8 @@ object SyncView {
       println("=======================================================================================")
       println(s"BEGINNING JOB $jobName FOR $currentDateString")
 
-      val ignoredCount = new LongAccumulator(); // Number of records ignored
-      val processedCount = new LongAccumulator();  // Number of Records Processed
+      val ignoredCount = sc.accumulator(0, "Number of Records Ignored")
```
Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/views/SyncView.scala]: try `sc.longAccumulator("Number of Records Ignored")`

**Comment by thomcc** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/SyncView.scala, line 474 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149#-KXuvmfS5x7fdg2b56tN:-KXvtPqjbvrxB5v2s2RE:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/4ba5c1ab150a3c223e853684b376b730874246e3/src/main/scala/com/mozilla/telemetry/views/SyncView.scala#L474)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary><blockquote>

Please change this comment to match the others, i.e.: s/required/a required field/. 
</blockquote></details>

Done.

---

*[src/main/scala/com/mozilla/telemetry/views/SyncView.scala, line 66 at r2](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149#-KXqpuTDT7svzfPrvW2u-r2-66:-KXvtRzSs4fHav8OP25J:by93e82) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/b67ce7fce4ceefa0598fdf8b864486628a41f04e/src/main/scala/com/mozilla/telemetry/views/SyncView.scala#L66)):*
<details><summary><i>Previously, mreid-moz (Mark Reid) wrote…</i></summary><blockquote>

try `sc.longAccumulator("Number of Records Ignored")`
</blockquote></details>

That did the trick, thanks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149)*

**Comment by thomcc** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/main/scala/com/mozilla/telemetry/views/SyncView.scala, line 66 at r2](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149#-KXqpuTDT7svzfPrvW2u-r2-66:-KXwbOzTWEi8g-vAcBTP:b-fu5vew) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/b67ce7fce4ceefa0598fdf8b864486628a41f04e/src/main/scala/com/mozilla/telemetry/views/SyncView.scala#L66)):*
<details><summary><i>Previously, thomcc (Thom) wrote…</i></summary><blockquote>

That did the trick, thanks.
</blockquote></details>

Done. (I guess I needed to press this button to resolve this?)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149)*

**Comment by thomcc** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/main/scala/com/mozilla/telemetry/views/SyncView.scala, line 66 at r2](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149#-KXqpuTDT7svzfPrvW2u-r2-66:-KXwbjXw4wNf81d_GEtj:b-eibssq) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/b67ce7fce4ceefa0598fdf8b864486628a41f04e/src/main/scala/com/mozilla/telemetry/views/SyncView.scala#L66)):*
<details><summary><i>Previously, thomcc (Thom) wrote…</i></summary><blockquote>

Done. (I guess I needed to press this button to resolve this?)
</blockquote></details>

Ah, nope, probably just waiting on mreid to do so, sorry!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/149)*

**Pull request merged**